src/cache/cache_las.c

```diff
@@ -765,46 +685,122 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
                 mementos_cnt++;
             }
 
-            /*
-             * If saving a non-zero length value on the page, save a birthmark instead of
-             * duplicating it in the lookaside table. (We check the length because row-store doesn't
-             * write zero-length data items.)
-             */
-            if (upd == list->onpage_upd && upd->size > 0) {
-                WT_ASSERT(
-                  session, upd->type == WT_UPDATE_STANDARD || upd->type == WT_UPDATE_MODIFY);
-                /* Make sure that we are generating a birthmark for an in-memory update. */
-                mementop->txnid = upd->txnid;
-                mementop->durable_ts = upd->durable_ts;
-                mementop->start_ts = upd->start_ts;
-                mementop->prepare_state = upd->prepare_state;
+            WT_ERR(__wt_modify_vector_push(&modifies, upd));
+        }
 
-                /*
-                 * Store the current update commit timestamp and transaction id, these are the stop
-                 * timestamp and transaction id's for the next record in the update list.
-                 */
-                stop_ts = upd->start_ts;
-                stop_txnid = upd->txnid;
+        upd = prev_upd = NULL;
+        squashed = false;
 
-                /* Do not put birthmarks into the lookaside. */
+        /*
+         * Get the oldest full update on chain. It is either the oldest update or the second oldest
+         * update if the oldest update is a TOMBSTONE and its txnid is WT_TXN_NONE and its timestamp
+         * is WT_TXN_NONE
+         */
+        WT_ASSERT(session, modifies.size > 0);
+        __wt_modify_vector_pop(&modifies, &upd);
+        WT_ASSERT(session, upd->type == WT_UPDATE_STANDARD ||
+            (upd->type == WT_UPDATE_TOMBSTONE && upd->txnid == WT_TXN_NONE &&
```

Comment:
When this scenario happens that `upd` with no txnid and timestamp until unless it is uncommitted?

Reply:
Added a comment